### DISCUSSION TITLE
Can controller watch/pull templates from git repo?

### DISCUSSION THREAD

**Discussion by mubarak-j** [argoproj/argo-workflows#9426] [Q&A] [answered]
Can workflow control watch/pull templates from git repo and update pipelines if changes detected? 
I’m trying to setup cicd pipeline and hoping to  avoid developers using cli to submit templates but instead push them to Git. So basically similar to ArgoCD using gitops pattern if that’s even possible?

**Comment by terrytangyuan** [comment, marked as answer]
You can use Argo Events to trigger new workflows (including updating templates). ArgoCD can be used to GitOps your templates as well.
